### ISSUE TITLE
Question: make js target not part of all

### ISSUE THREAD

**Issue by EmixamPP** [saghul/txiki.js#561]
Is there a reason why the `make js` target is not part of `all`?
And consequently, by extension, why including the precompiled js code (i.e. `src/bundles/c`) in the repo instead of creating them during the compilation of txiki.js.

It creates more manual steps during the development, make the PRs management more difficult (rebasing/merging becomes painful). And that can be "dangerous". If you forgot to execute `make js` on your latest changes, nothing will prevent the merge on master, and you will end up with an inconsistent source. 

I'm just curious to know the reason behind that

**Comment by lal12** [comment]
I think the reasoning was not to require the npm stuff as hard requirements for building. Though I am not the biggest fan of this either.

**Comment by lal12** [comment]
I had the source distribution in mind before as well. That would work analog to the classical automake way.

Would be easy to include that in CI/CD.

**Comment by saghul** [comment]
Out of curiosity, other than the git churn, what are the pain points the current approach has in your case?

**Comment by EmixamPP** [comment]
Here is my use case (maybe niche), I cross-compile `txiki.js` for arm platform, and I extend it with some feature specific to my project, thus I have a set of patches to apply on the repo.
However, since the patches alter the js, either they must contain also the diffs for the pre-compiled bundled code, or the build steps have to execute `make js` after applying the patch.
The first solution is very annoying because it does not allow to keep a patch compatible between the different release of `txiki.js`, the second one is better, but when making the patch, I have to make sure to not include the changes of `make js`, and when applying it manually for example, I shouldn't forget to execute `make js` _then_ `make`.

Anyway, there is no big deal, the main "danger" I see, is to end up with an inconsistent source. But I totally understand now the choice of including these bundled files into the repo.

Thinking of that, when I started playing with the repo, it wasn't clear to me that I had to execute `make js` when altering the `js` code. So, I wish I'd had a line or two of documentation explaining what to do.
